DT not handling a blank name in an SBOM

### Current Behavior

In theory, this should never happen, but.....

Tooling that automatically processes open source packages will return the package name based on a variety of possible fields in package definition files. Some of these have been incorrectly filled out and have a blank name.

The tooling puts a blank name in the field of the SBOM such as

`"name": "",`

As the field is there, it does not fail on the validation of the SBOM schema as it is a required field for the component, technically it does not require content.

When this imports into DT it gives the following:

```
[36mdtrack-apiserver_1  |[0m 2023-06-14 13:09:21,258 WARN [Persist] Insert of object "org.dependencytrack.model.Component@430b2e5f" using statement "INSERT INTO COMPONENT (AUTHOR,BLAKE2B_256,BLAKE2B_384,BLAKE2B_512,BLAKE3,CLASSIFIER,COPYRIGHT,CPE,DESCRIPTION,DIRECT_DEPENDENCIES,EXTENSION,EXTERNAL_REFERENCES,FILENAME,"GROUP",INTERNAL,LAST_RISKSCORE,LICENSE,LICENSE_URL,MD5,"NAME",TEXT,PARENT_COMPONENT_ID,PROJECT_ID,PUBLISHER,PURL,PURLCOORDINATES,LICENSE_ID,SHA1,SHA_256,SHA_384,SHA3_256,SHA3_384,SHA3_512,SHA_512,SWIDTAGID,UUID,VERSION) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)" failed : NULL not allowed for column "NAME"; SQL statement:
[36mdtrack-apiserver_1  |[0m INSERT INTO COMPONENT (AUTHOR,BLAKE2B_256,BLAKE2B_384,BLAKE2B_512,BLAKE3,CLASSIFIER,COPYRIGHT,CPE,DESCRIPTION,DIRECT_DEPENDENCIES,EXTENSION,EXTERNAL_REFERENCES,FILENAME,"GROUP",INTERNAL,LAST_RISKSCORE,LICENSE,LICENSE_URL,MD5,"NAME",TEXT,PARENT_COMPONENT_ID,PROJECT_ID,PUBLISHER,PURL,PURLCOORDINATES,LICENSE_ID,SHA1,SHA_256,SHA_384,SHA3_256,SHA3_384,SHA3_512,SHA_512,SWIDTAGID,UUID,VERSION) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) [23502-214]
[36mdtrack-apiserver_1  |[0m 2023-06-14 13:09:21,259 ERROR [BomUploadProcessingTask] Error while processing bom
[36mdtrack-apiserver_1  |[0m javax.jdo.JDODataStoreException: Insert of object "org.dependencytrack.model.Component@430b2e5f" using statement "INSERT INTO COMPONENT (AUTHOR,BLAKE2B_256,BLAKE2B_384,BLAKE2B_512,BLAKE3,CLASSIFIER,COPYRIGHT,CPE,DESCRIPTION,DIRECT_DEPENDENCIES,EXTENSION,EXTERNAL_REFERENCES,FILENAME,"GROUP",INTERNAL,LAST_RISKSCORE,LICENSE,LICENSE_URL,MD5,"NAME",TEXT,PARENT_COMPONENT_ID,PROJECT_ID,PUBLISHER,PURL,PURLCOORDINATES,LICENSE_ID,SHA1,SHA_256,SHA_384,SHA3_256,SHA3_384,SHA3_512,SHA_512,SWIDTAGID,UUID,VERSION) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)" failed : NULL not allowed for column "NAME"; SQL statement:
[36mdtrack-apiserver_1  |[0m INSERT INTO COMPONENT (AUTHOR,BLAKE2B_256,BLAKE2B_384,BLAKE2B_512,BLAKE3,CLASSIFIER,COPYRIGHT,CPE,DESCRIPTION,DIRECT_DEPENDENCIES,EXTENSION,EXTERNAL_REFERENCES,FILENAME,"GROUP",INTERNAL,LAST_RISKSCORE,LICENSE,LICENSE_URL,MD5,"NAME",TEXT,PARENT_COMPONENT_ID,PROJECT_ID,PUBLISHER,PURL,PURLCOORDINATES,LICENSE_ID,SHA1,SHA_256,SHA_384,SHA3_256,SHA3_384,SHA3_512,SHA_512,SWIDTAGID,UUID,VERSION) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) [23502-214]
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.api.jdo.JDOAdapter.getJDOExceptionForNucleusException(JDOAdapter.java:605)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.api.jdo.JDOPersistenceManager.jdoMakePersistent(JDOPersistenceManager.java:702)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.api.jdo.JDOPersistenceManager.makePersistent(JDOPersistenceManager.java:722)
[36mdtrack-apiserver_1  |[0m 	at alpine.persistence.AbstractAlpineQueryManager.persist(AbstractAlpineQueryManager.java:427)
[36mdtrack-apiserver_1  |[0m 	at org.dependencytrack.persistence.ComponentQueryManager.createComponent(ComponentQueryManager.java:306)
[36mdtrack-apiserver_1  |[0m 	at org.dependencytrack.persistence.QueryManager.createComponent(QueryManager.java:515)
[36mdtrack-apiserver_1  |[0m 	at org.dependencytrack.tasks.BomUploadProcessingTask.processComponent(BomUploadProcessingTask.java:206)
[36mdtrack-apiserver_1  |[0m 	at org.dependencytrack.tasks.BomUploadProcessingTask.inform(BomUploadProcessingTask.java:141)
[36mdtrack-apiserver_1  |[0m 	at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:101)
[36mdtrack-apiserver_1  |[0m 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[36mdtrack-apiserver_1  |[0m 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[36mdtrack-apiserver_1  |[0m 	at java.base/java.lang.Thread.run(Unknown Source)
[36mdtrack-apiserver_1  |[0m Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: NULL not allowed for column "NAME"; SQL statement:
[36mdtrack-apiserver_1  |[0m INSERT INTO COMPONENT (AUTHOR,BLAKE2B_256,BLAKE2B_384,BLAKE2B_512,BLAKE3,CLASSIFIER,COPYRIGHT,CPE,DESCRIPTION,DIRECT_DEPENDENCIES,EXTENSION,EXTERNAL_REFERENCES,FILENAME,"GROUP",INTERNAL,LAST_RISKSCORE,LICENSE,LICENSE_URL,MD5,"NAME",TEXT,PARENT_COMPONENT_ID,PROJECT_ID,PUBLISHER,PURL,PURLCOORDINATES,LICENSE_ID,SHA1,SHA_256,SHA_384,SHA3_256,SHA3_384,SHA3_512,SHA_512,SWIDTAGID,UUID,VERSION) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) [23502-214]
[36mdtrack-apiserver_1  |[0m 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)
[36mdtrack-apiserver_1  |[0m 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
[36mdtrack-apiserver_1  |[0m 	at org.h2.message.DbException.get(DbException.java:223)
[36mdtrack-apiserver_1  |[0m 	at org.h2.message.DbException.get(DbException.java:199)
[36mdtrack-apiserver_1  |[0m 	at org.h2.table.Column.validateConvertUpdateSequence(Column.java:365)
[36mdtrack-apiserver_1  |[0m 	at org.h2.table.Table.convertInsertRow(Table.java:926)
[36mdtrack-apiserver_1  |[0m 	at org.h2.command.dml.Insert.insertRows(Insert.java:167)
[36mdtrack-apiserver_1  |[0m 	at org.h2.command.dml.Insert.update(Insert.java:135)
[36mdtrack-apiserver_1  |[0m 	at org.h2.command.CommandContainer.executeUpdateWithGeneratedKeys(CommandContainer.java:242)
[36mdtrack-apiserver_1  |[0m 	at org.h2.command.CommandContainer.update(CommandContainer.java:163)
[36mdtrack-apiserver_1  |[0m 	at org.h2.command.Command.executeUpdate(Command.java:252)
[36mdtrack-apiserver_1  |[0m 	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
[36mdtrack-apiserver_1  |[0m 	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:169)
[36mdtrack-apiserver_1  |[0m 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
[36mdtrack-apiserver_1  |[0m 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.store.rdbms.SQLController.executeStatementUpdate(SQLController.java:430)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.store.rdbms.request.InsertRequest.execute(InsertRequest.java:520)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.insertObjectInTable(RDBMSPersistenceHandler.java:162)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.insertObject(RDBMSPersistenceHandler.java:138)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.state.StateManagerImpl.internalMakePersistent(StateManagerImpl.java:4587)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.state.StateManagerImpl.makePersistent(StateManagerImpl.java:4564)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.ExecutionContextImpl.persistObjectInternal(ExecutionContextImpl.java:2014)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.ExecutionContext.persistObjectInternal(ExecutionContext.java:320)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.ExecutionContextImpl.persistObjectWork(ExecutionContextImpl.java:1862)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.ExecutionContextImpl.persistObject(ExecutionContextImpl.java:1723)
[36mdtrack-apiserver_1  |[0m 	at org.datanucleus.api.jdo.JDOPersistenceManager.jdoMakePersistent(JDOPersistenceManager.java:697)
[36mdtrack-apiserver_1  |[0m 	... 10 common frames omitted
[36mdtrack-apiserver_1  |[0m 2023-06-14 13:09:21,268 WARN [General] ExecutionContext closed with active transaction, so rolling back the active transaction

```

At this point it stops importing any further entries and does not process any vulnerabilities in those that have been imported.

The only way to tell there is an error is to look in the logs, but if you only have access to the web interface all appears to have worked, just nothing is happening after it has failed.

In theory, you could skip this entry and let the rest continue and process properly, but wondering if the SBOM should just completely reject it with an error for them to fix.

### Steps to Reproduce

1.Import an SBOM with a blank component (library) name

### Expected Behavior

Not error in the way that it does

### Dependency-Track Version

4.8.0

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

Whatever is in the container.

### Browser

Mozilla Firefox

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

--- a/src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java
+++ b/src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java
@@ -75,6 +75,7 @@ import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 import java.util.function.Function;
 
+import static java.util.Objects.requireNonNullElse;
 import static org.apache.commons.lang3.StringUtils.isNotBlank;
 import static org.apache.commons.lang3.StringUtils.trim;
 import static org.apache.commons.lang3.StringUtils.trimToNull;
@@ -167,7 +168,7 @@ public class ModelConverter {
         component.setSupplier(convert(cdxComponent.getSupplier()));
         component.setClassifier(convertClassifier(cdxComponent.getType()).orElse(Classifier.LIBRARY));
         component.setGroup(trimToNull(cdxComponent.getGroup()));
-        component.setName(trimToNull(cdxComponent.getName()));
+        component.setName(requireNonNullElse(trimToNull(cdxComponent.getName()), "-"));
         component.setVersion(trimToNull(cdxComponent.getVersion()));
         component.setDescription(trimToNull(cdxComponent.getDescription()));
         component.setCopyright(trimToNull(cdxComponent.getCopyright()));
@@ -325,7 +326,7 @@ public class ModelConverter {
         final var service = new ServiceComponent();
         service.setBomRef(useOrGenerateRandomBomRef(cdxService.getBomRef()));
         service.setGroup(trimToNull(cdxService.getGroup()));
-        service.setName(trimToNull(cdxService.getName()));
+        service.setName(requireNonNullElse(trimToNull(cdxService.getName()), "-"));
         service.setVersion(trimToNull(cdxService.getVersion()));
         service.setDescription(trimToNull(cdxService.getDescription()));
         service.setAuthenticated(cdxService.getAuthenticated());

--- a/src/test/java/org/dependencytrack/tasks/BomUploadProcessingTaskTest.java
+++ b/src/test/java/org/dependencytrack/tasks/BomUploadProcessingTaskTest.java
@@ -1065,6 +1065,45 @@ public class BomUploadProcessingTaskTest extends PersistenceCapableTest {
         });
     }
 
+    @Test
+    public void informWithEmptyComponentAndServiceNameTest() {
+        final var project = new Project();
+        project.setName("acme-license-app");
+        qm.persist(project);
+
+        final byte[] bomBytes = """
+                {
+                  "bomFormat": "CycloneDX",
+                  "specVersion": "1.4",
+                  "serialNumber": "urn:uuid:3e671687-395b-41f5-a30f-a58921a69b80",
+                  "version": 1,
+                  "components": [
+                    {
+                      "type": "library",
+                      "name": ""
+                    }
+                  ],
+                  "services": [
+                    {
+                      "name": ""
+                    }
+                  ]
+                }
+                """.getBytes(StandardCharsets.UTF_8);
+
+        final var bomUploadEvent = new BomUploadEvent(qm.detach(Project.class, project.getId()), bomBytes);
+        new BomUploadProcessingTask().inform(bomUploadEvent);
+        awaitBomProcessedNotification(bomUploadEvent);
+
+        qm.getPersistenceManager().evictAll();
+        assertThat(qm.getAllComponents(project)).satisfiesExactly(component -> {
+            assertThat(component.getName()).isEqualTo("-");
+        });
+        assertThat(qm.getAllServiceComponents(project)).satisfiesExactly(service -> {
+            assertThat(service.getName()).isEqualTo("-");
+        });
+    }
+
     @Test // https://github.com/DependencyTrack/dependency-track/issues/1905
     public void informIssue1905Test() throws Exception {
         final var project = qm.createProject("Acme Example", null, "1.0", null, null, null, true, false);
@@ -1331,7 +1370,6 @@ public class BomUploadProcessingTaskTest extends PersistenceCapableTest {
 
     @Test
     public void informIssue3936Test() throws Exception{
-        
         final Project project = qm.createProject("Acme Example", null, "1.0", null, null, null, true, false);
         List<String> boms = new ArrayList<>(Arrays.asList("/unit/bom-issue3936-authors.json", "/unit/bom-issue3936-author.json", "/unit/bom-issue3936-both.json"));
         for(String bom : boms){

